File: src/Lists/Node.java
package Lists;

class Node {
    public String content;
    public Node next;
    public Node prev;

    public Node(String content) {
        this.content = content;
        this.next = null;
        this.prev = null;
    }

    public String toString() {
        return this.content;
    }
}

File: src/com/arainko/redblacktrees/Node.kt
package com.arainko.redblacktrees

import com.arainko.redblacktrees.helpers.NodeField.*
import com.arainko.redblacktrees.helpers.Color
import com.arainko.redblacktrees.helpers.NodeField

class Node(val content: Int) : Comparable<Node> {

    var parent: NodeField = Empty
    var leftChild: NodeField = Empty
    var rightChild: NodeField = Empty
    lateinit var color: Color

    override fun compareTo(other: Node): Int = this.content - other.content

    override fun toString(): String = "Node($content)"
}